backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java

```diff
@@ -46,17 +49,46 @@ void init() throws NumberParseException {
 
   @AuthorizationConfiguration.RequirePermissionStartTestWithPatientLink
   @Transactional(noRollbackFor = {TwilioException.class, ApiException.class})
-  public String sendToPatientLink(UUID patientLinkId, String text) throws NumberParseException {
+  public List<SmsDeliveryResult> sendToPatientLink(UUID patientLinkId, String text) {
     PatientLink pl = pls.getRefreshedPatientLink(patientLinkId);
-    String messageId = sendToPerson(pl.getTestOrder().getPatient(), text);
-    tmsRepo.save(new TextMessageSent(pl, messageId));
-    return messageId;
+
+    List<SmsDeliveryResult> smsSendResults = sendToPerson(pl.getTestOrder().getPatient(), text);
+
+    smsSendResults.forEach(
+        smsDeliveryResult -> {
+          if (!smsDeliveryResult.getDeliverySuccess()) {
+            return;
+          }
+
+          tmsRepo.save(new TextMessageSent(pl, smsDeliveryResult.getMessageId()));
+        });
+
+    return smsSendResults;
   }
 
-  private String sendToPerson(Person p, String text) throws NumberParseException {
-    String msgId = sms.send(new PhoneNumber(formatNumber(p.getTelephone())), fromNumber, text);
-    LOG.debug("SMS send initiated {}", msgId);
-    return msgId;
+  private List<SmsDeliveryResult> sendToPerson(Person p, String text) {
+    List<SmsDeliveryResult> smsSendResults = new ArrayList<>();
+
+    p.getPhoneNumbers().stream()
```

Comment:
**question:** should this be filtered down to only the `MOBILE` type of phone numbers?

Reply:
Inspecting the acceptance criteria more closely 🕵️‍♂️, they do in fact say to:

> [...] update the question text to asks users if they would like to text all or none of the **mobile numbers.**

So, I'm going to say "yes". @aliciabeckett-gov can you confirm? It would certainly make sense.

Reply:
Done in c5a4c27.